### ISSUE TITLE
AWS setup script doesn't wait for S3 bucket to be ready

### ISSUE THREAD

**Issue by amitkgupta** [kubernetes/kubernetes#12109]
When I run:

```
$ export KUBERNETES_PROVIDER=aws; curl -sS https://get.k8s.io | bash
```

I get the following error:

```
....
make_bucket: s3://kubernetes-staging-SOME_PROJECT_HASH/
+++ Staging server tars to S3 Storage: kubernetes-staging-SOME_PROJECT_HASH/devel
upload: ../../../../var/folders/yg/nckvbn8s7810t6rc_j9g43cr0000gn/T/kubernetes.XXXXXX.zFJyjpGo/s3/kubernetes-salt.tar.gz to s3://kubernetes-staging-SOME_PROJECT_HASH/devel/kubernetes-salt.tar.gz
upload failed: ../../../../var/folders/yg/nckvbn8s7810t6rc_j9g43cr0000gn/T/kubernetes.XXXXXX.zFJyjpGo/s3/kubernetes-server-linux-amd64.tar.gz to s3://kubernetes-staging-SOME_PROJECT_HASH/devel/kubernetes-server-linux-amd64.tar.gz
A client error (NoSuchBucket) occurred when calling the UploadPart operation: The specified bucket does not exist
```

Waiting a bit and running `./cluster/kube-up.sh` got past this point because the bucket was ready the second time around, but the getting-started process should be more robust.

**Comment by drewblas** [comment]
I experienced the same issue

**Comment by justinsb** [comment]
We do try to query for the bucket existing, but perhaps S3 is even more eventually consistent than I thought!  https://github.com/GoogleCloudPlatform/kubernetes/blob/f48543aba54e7edd545389fd104695463f0a354b/cluster/aws/util.sh#L498-L514

I did observe that we were very trusting about the user providing the correct S3 region, and have tried to fix this in #12125.  I don't think that was the problem here though (if you make_bucket, you can't the version wrong...)

I guess we could retry the sync a few times; might help with e.g. a network interruption during the sync also.

**Comment by justinsb** [comment]
I believe this to have been fixed by #12125.  I'm going to close, but if it happens again please do reopen.

**Comment by matthewkmayer** [comment]
FYI I got this with the latest k8s as of 10/13/15:

```bash
Uploading to Amazon S3
Creating kubernetes-staging-cf9c00441beff5d609b0a01527c980db
make_bucket: s3://kubernetes-staging-cf9c00441beff5d609b0a01527c980db/
+++ Staging server tars to S3 Storage: kubernetes-staging-cf9c00441beff5d609b0a01527c980db/devel
upload failed: ../../../../../../var/folders/wv/y4xfrd3501sg9cltcbp_04tw0000gn/T/kubernetes.XXXXXX.Xt1cQsik/s3/kubernetes-server-linux-amd64.tar.gz to s3://kubernetes-staging-cf9c00441beff5d609b0a01527c980db/devel/kubernetes-server-linux-amd64.tar.gz
A client error (NoSuchBucket) occurred when calling the UploadPart operation: The specified bucket does not exist
upload: ../../../../../../var/folders/wv/y4xfrd3501sg9cltcbp_04tw0000gn/T/kubernetes.XXXXXX.Xt1cQsik/s3/kubernetes-salt.tar.gz to s3://kubernetes-staging-cf9c00441beff5d609b0a01527c980db/devel/kubernetes-salt.tar.gz
```

I believe this is k8s 1.0.6 but I just tore down my AWS cluster and apparently kubectl needs to talk to the now-terminated master instance.  

I ran

```bash
curl -sS https://get.k8s.io | bash
``` 
to get k8s.

Waiting and running `kube-up` again doesn't show the issue since the bucket is fully created and accessible.

I just provisioned a Vagrant cluster in hopes that would help show what version I'm using.  Here's the output of `kube-ctl version`:

```bash
Client Version: version.Info{Major:"1", Minor:"0", GitVersion:"v1.0.6", GitCommit:"388061f00f0d9e4d641f9ed4971c775e1654579d", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"0", GitVersion:"v1.0.6", GitCommit:"388061f00f0d9e4d641f9ed4971c775e1654579d", GitTreeState:"clean"}
```
